# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1711565207 -3600
# Parent  d142293630ae4bd05ed2ca542d7be7f056532a80
Feature: pref mail.biff.show_for_server to disable biff display in tree for servers.

See https://bugzilla.mozilla.org/show_bug.cgi?id=1545955#c5 for some details.

Also suppress biff for Gmail's "All Mail" folder: https://bugzilla.mozilla.org/show_bug.cgi?id=1867223

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -2354,8 +2354,21 @@ var folderPane = {
    * @param {nsIMsgFolder} folder
    * @param {boolean} hasNewMessages
    */
   changeNewMessages(folder, hasNewMessages) {
+    if (
+      folder.isServer &&
+      !Services.prefs.getBoolPref("mail.biff.show_for_server", true)
+    ) {
+      return;
+    }
+    if (
+      folder.flags & Ci.nsMsgFolderFlags.Archive &&
+      folder.server instanceof Ci.nsIImapIncomingServer &&
+      folder.server.isGMailServer
+    ) {
+      return;
+    }
     this._changeRows(folder, row => {
       // Find the nearest visible ancestor and update it.
       let collapsedAncestor = row.parentElement?.closest("li.collapsed");
       while (collapsedAncestor) {
diff --git a/mailnews/base/src/FolderUtils.sys.mjs b/mailnews/base/src/FolderUtils.sys.mjs
--- a/mailnews/base/src/FolderUtils.sys.mjs
+++ b/mailnews/base/src/FolderUtils.sys.mjs
@@ -78,8 +78,9 @@ function getFolderProperties(aFolder, aO
 
   // set the SpecialFolder attribute
   properties.push("specialFolder-" + getSpecialFolderString(aFolder));
 
+  /* Dead code, not used in 115 any more.
   // Now set the biffState
   switch (aFolder.biffState) {
     case Ci.nsIMsgFolder.nsMsgBiffState_NewMail:
       properties.push("biffState-NewMail");
@@ -89,8 +90,9 @@ function getFolderProperties(aFolder, aO
       break;
     default:
       properties.push("biffState-UnknownMail");
   }
+  */
 
   properties.push("isSecure-" + aFolder.server.isSecure);
 
   // A folder has new messages, or a closed folder or any subfolder has new messages.
diff --git a/mailnews/mailnews.js b/mailnews/mailnews.js
--- a/mailnews/mailnews.js
+++ b/mailnews/mailnews.js
@@ -899,8 +899,10 @@ pref("mail.biff.use_new_count_in_badge",
 #ifdef XP_WIN
 pref("mail.biff.use_new_count_in_badge", true);
 #endif
 
+pref("mail.biff.show_for_server", true);
+
 // For feed account serverType=rss sound on biff.
 // Allow for a different sound to be played for new feed articles.
 pref("mail.feed.play_sound", false);
 pref("mail.feed.play_sound.type", 0);
